lib/config/schemas/gateway.config.json

```diff
@@ -107,6 +107,9 @@
         "port": {
           "type": "number"
         },
+        "host": {
+          "type": "string"
+        },
```

Comment:
You should update the JSON Schema to make sure only one of the two properties are present (host or hostname)

Reply:
- If we remove `hostname` from here, will it still be able to support old configurations?
- Btw, what's the purpose of this JSON Schema? I'm not sure of this

Reply:
We're not going to remove it, we're going to put a constraint so that the user will only use `host` or `hostname`, using the JSON Schema.

The JSON Schema is basically a JSON file telling how an object should look like, so that you can validate it and make actions.

For instance, if I declare a JSON Schema like this:

```json
{
  "$id": "http://express-gateway.io/models/users.json",
  "type": "object",
  "properties": {
    "firstname": {
      "type": "string"
    },
    "lastname": {
      "type": "string"
    },
    "username": {
      "type": "string"
    },
    "email": {
      "type": "string",
      "format": "email"
    },
    "redirectUri": {
      "type": "string",
      "format": "uri"
    }
  },
  "required": [
    "username",
    "firstname",
    "lastname"
  ]
}
```

Then the object passing the validation will have al least     "username","firstname", "lastname"  properties defined and they're a string.

We use that to validate the configuration files `gateway.config.yml` and `system.config.yml`

Reply:
Got it, thanks!

Reply:
Once addressed, do not forget to mark the conversation as resolved. This helps the reviews ;)

<img width="313" alt="image" src="https://user-images.githubusercontent.com/1416224/54291401-c9922380-45ac-11e9-8d4d-b74deb02af86.png">